enhancements/installer/bootstrapping-clusters-with-capi-providers.md

```diff
@@ -0,0 +1,646 @@
+
+The changes proposed in this enhancement allow the OpenShift Installer to change the underlying mechanics on how an OpenShift cluster is created from an InstallConfig. The change of technology carries structural differences in how the infrastructure is provisioned, defaults, or the topology of the cluster.
+
+Given an InstallConfig to both systems the enhancement guarantees a functionally equivalent OpenShift cluster.
+
+In the best case scenario, OpenShift users rely on tags or labels attached to cloud resources when automating specific parts of their infrastructure. While this can be true in most cases, a goal of this proposal is to document in details the underlying structural differences.
+
+### Risks and Mitigations
+
+While we do not expect these changes to introduce a significant security risk, we are working with product security teams to ensure they are aware of the changes and are able to review.
+
+Each Cluster API provider's created infrastructure should be reviewed by product security and subject area experts to ensure the new infrastructure topologies considered are on par with today's standards.
+
+### Drawbacks
+
+Users will be required to edit additional files if they are manually editing control-plane machine manifests.
+Changes would need to be made to both the machine-api and cluster-api manifests.
+
+#### External/Upstream dependencies
+
+As explained above, the Installer will embed binaries for each supported CAPI provider, as well
+as `etcd` and `kube-apiserver` binaries. Initially, CAPI providers will be vendored and built in
+the Installer repo. `etcd` and `kube-apiserver` will be copied from the release image.
+
+##### In-tree Dependency Vendoring
+
+For the initial implementation, CAPI providers will be vendored and built within the Installer repo:
+
+```bash
+$ tree cluster-api/providers -L 2
+cluster-api/providers
+├── aws
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+├── azure
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+...
+```
+
+This follows a similar pattern to the current Terraform provider implementation and build. This
+pattern will be used initially due to its simplicity and existing support. This pattern has
+drawbacks because changes to providers need to be merged upstream first and then vendored
+to the Installer. This aspect will be particularly problematic once CAPI providers are GA
+and we need to keep the Installer in-sync with other providers. Builds are also inefficient
+in that providers are always rebuilt, even when unchanged.
+
+##### Copying Dependencies from Container Images
+
+[TODO]
+
+#### Disk space and memory
+
+> NOTE: All tests are performed on a MacOS system at the time of writing.
+
+With the introduction of the Local Control Plane, new required components are being run during cluster creation.
+
+- `api-server` uses ~280MB and `etcd` ~52MB
```

Comment:
Not clear, is this disk space, or memory, may want to be explicit

Reply:
added a note